### PR TITLE
Update operator framework to operator-sdk v1.15

### PR DESCRIPTION
**Pull request by mrogers950**
Notable adjustments:

- Makefile overhaul: Most development targets have changed, and have a workflow
  based on the IMAGE_REPO and TAG environment variables, see README.md. The
  local image-to-cluster targets have been removed.
- The deploy/ directory has been replaced by config/ and bundle/ which are
  updated by generate, build, and deploy targets.
- Deprecate RELATED_IMAGE_OPERATOR use inside of the operator. The value will
  always resolve to the same image as the operator deployment to ensure the
  daemonSet pod is aligned.
- Remove all assumptions of a `openshift-file-integrity` operator namespace in
  the controller metrics setup.
- E2E tests run in a test-specific namespace and properly clean resources.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -66,331 +64,348 @@ E2E_USE_DEFAULT_IMAGES?=false
...
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths=./pkg/apis/fileintegrity/v1alpha1
+
+
+##@ Build
+
+.PHONY: all
+all: images
+
+.PHONY: images
+images: image bundle-image  ## Build operator and bundle images.
```
Comment (rhmdnd) [inline Makefile:267]: I think we lost the `image` target in this update.

```
$ grep -R '.PHONY: image' Makefile
.PHONY: images
```

Comment (rhmdnd) [reply Makefile:267]: Nevermind - I found it.

File: Makefile [hunk 2]
```diff
@@ -66,331 +63,348 @@ E2E_USE_DEFAULT_IMAGES?=false
...
+
+.PHONY: operator-sdk
+SDK_BIN = ./build/operator-sdk
+operator-sdk: ## Download operator-sdk locally if necessary.
+ifeq (,$(wildcard $(SDK_BIN)))
+ifeq (,$(shell which operator-sdk 2>/dev/null))
+	@{ \
+	set -e ;\
+	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
+	curl -sSLo $(SDK_BIN) https://github.com/operator-framework/operator-sdk/releases/download/v1.15.0/operator-sdk_$${OS}_$${ARCH} ;\
```
[the remark this answers is not in the document]
Comment (rhmdnd) [reply Makefile]: We can do this in a follow-up, too. There's a lot going on in this patch already.

**Comment by rhmdnd** [comment]
I was able to get a clean e2e run on the latest commit. Waiting for CI, but I think I'm OK with merging this and cleaning up any remaining bits in follow-on patches.

File: bundle.Dockerfile [hunk 3]
```diff
@@ -1,11 +1,21 @@
 FROM scratch
 
+# Core bundle labels.
 LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
 LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=file-integrity-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
 LABEL operators.operatorframework.io.bundle.channel.default.v1=alpha
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.14.0+git
```
[the remark this answers is not in the document]
Comment (mrogers950) [reply bundle.Dockerfile:10]: That's actually the version the v1.15 operator-sdk stages with its generated commands

File: go.mod [hunk 4]
```diff
@@ -1,66 +1,34 @@
...
-	github.com/go-openapi/spec v0.19.4
-	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
-	github.com/openshift/library-go v0.0.0-20210708191609-4b9033d00d37
+	github.com/mitchellh/go-homedir v1.1.0
+	github.com/onsi/ginkgo v1.16.4
+	github.com/onsi/gomega v1.18.1
+	github.com/openshift/library-go v0.0.0-20200831114015-2ab0c61c15de
 	github.com/openshift/machine-config-operator v0.0.1-0.20200913004441-7eba765c69c9
-	github.com/operator-framework/operator-registry v1.13.4
-	github.com/operator-framework/operator-sdk v0.19.4
```
[the remark this answers is not in the document]
Comment (mrogers950) [reply go.mod:16]: nope!

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrogers950) [reply Makefile]: did that

**Comment by mrogers950** [comment]
I'll do a cleanup pass and squash the commits.

File: cmd/manager/operator.go [hunk 5]
```diff
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package main
+package manager
```
Comment (rhmdnd) [inline cmd/manager/operator.go:16]: The s/main/manager/ bit is a new expectation of the operator-sdk, right?

File: config/manifests/bases/file-integrity-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -24,20 +24,21 @@ metadata:
     operatorframework.io/suggested-namespace: openshift-file-integrity
     operators.openshift.io/infrastructure-features: '["disconnected", "fips"]'
     repository: https://github.com/openshift/file-integrity-operator
-  name: file-integrity-operator.v0.1.24
+  name: file-integrity-operator.v0.1.22
```
Comment (rhmdnd) [inline config/manifests/bases/file-integrity-operator.clusterserviceversion.yaml:27]: How come this was rolled back?

File: go.mod (hunk 4 above)
[the remark this answers is not in the document]
Comment (rhmdnd) [reply go.mod:16]: Just for completeness - where do we specify it now?

**Review by rhmdnd — COMMENTED** [review]
Thanks for separating out the vendor updates. So far, things look good to me but I'm mostly relying on the e2e tests to make sure things are still functional with a patch this large.

I'd like some else familiar with the code to give this a review, too.

File: config/manifests/bases/file-integrity-operator.clusterserviceversion.yaml (hunk 6 above)
Comment (mrogers950) [reply config/manifests/bases/file-integrity-operator.clusterserviceversion.yaml:27]: this file is only used as a template for generation (into the properly versioned manifest under bundle/)

File: cmd/manager/operator.go (hunk 5 above)
Comment (mrogers950) [reply cmd/manager/operator.go:16]: Yeah, here it doesn't matter so much, just aligns with how the framework names this part of an operator.

**Comment by mrogers950** [comment]
Second green run. Going for a 3rd.

/test e2e-aws

**Comment by JAORMX** [comment]
Looking forward to the compliance operator PR

**Comment by JAORMX** [comment]
Looking forward to the compliance operator PR

**Comment by rhmdnd** [comment]
> @mrogers950: The following test failed, say /retest to rerun all failed tests or /retest-required to rerun all mandatory failed tests:

Looks like the test failed to taint a node, and that caused a snowball of failures.

**Review by rhmdnd — APPROVED** [review]

**Pull request merged**